src/components/EventForm/EventForm.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useState, useCallback } from 'react';
+import { RouteComponentProps } from '@reach/router';
+import { formatISO, format } from 'date-fns';
+
+import Alert from 'src/components/Alert';
+import Button from 'src/components/inputs/Button';
+import DateField from 'src/components/fields/DateField';
+import SelectField from 'src/components/fields/SelectField';
+import TextAreaField from 'src/components/fields/TextAreaField';
+import TextField from 'src/components/fields/TextField';
+import TimeField from 'src/components/fields/TimeField';
+import { EventFormWrapper, Form } from './styled';
+import { EVENT_OPTIONS } from './constants';
+
+export type EventDataInput = {
+  id?: string;
+  name?: string;
+  description?: string;
+  startTime?: Date;
+  endTime?: Date;
+  imageUrl?: string;
+  imageAlt?: string;
+  location?: string;
+  eventType?: string;
+};
+
+export type EventDataOutput = {
+  id?: string;
+  name: string;
+  description: string;
+  startTime: Date;
+  endTime: Date;
+  imageUrl?: string;
+  imageAlt?: string;
+  location?: string;
+  eventType: string;
+};
+
+type EventFormProps = {
+  headerTitle: string;
+  onSubmit: (data: EventDataOutput) => Promise<void>;
+} & EventDataInput;
+
+function EventForm({ onSubmit, ...rest }: EventFormProps) {
+  const [name, setName] = useState(rest.name || '');
+  const [imageUrl, setImageUrl] = useState(rest.imageUrl || '');
+  const [imageAlt, setImageAlt] = useState(rest.imageAlt || '');
+  const [date, setDate] = useState(
+    format(
+      new Date((rest.startTime || new Date(Date.now())).toDateString()) ||
+        Date.now(),
```

Comment:
This expression appears unnecessary complex.
```suggestion
        rest.startTime == null ? Date.now() : new Date(rest.startTime)
```

Also, we should also try to use `date-fns` instead of the native `Date` methods. The native methods have a lot of quirky behavior, so it's better to just stay on the safe side and use the library whenever possible 🙈. Assuming that `rest.startTime` is an ISO timestamp, this means that we could use the `parseISO` function from `date-fns`.